src/main/java/com/cloudogu/smeagol/Dispatcher.java

```diff
@@ -0,0 +1,33 @@
+package com.cloudogu.smeagol;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * The dispatcher decides how to handle a request.
+ */
+public interface Dispatcher {
+
+    /**
+     * Returns {@code true} if the requested uri needs to be dispatched.
+     *
+     * @param uri requested uri
+     *
+     * @return {@code true} if the requested uri needs to be dispatched
+     */
+    boolean needsToBeDispatched(String uri);
+
+    /**
+     * Dispatches the request
+     *
+     * @param request http servlet request
+     * @param response http servlet response
+     * @param uri requested uri
+     *
+     * @throws ServletException servlet exception
+     * @throws IOException io exception
+     */
+    void dispatch(HttpServletRequest request, HttpServletResponse response, String uri) throws ServletException, IOException;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: javax.servlet.ServletException, java.io.IOException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1160)